jscomp/others/belt_String.ml

```diff
@@ -0,0 +1,161 @@
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
+
+(** JavaScript String API *)
+
+type t = string
+
+
+(** [concat append original] returns a new string with [append] added after [original].
+
+@example {[
+  concat "a" "b" = "ab";;
+]}
+*)
+external concat : t -> t -> t = "" [@@bs.send]
+
+(** [concat arr original] returns a new string consisting of each item of an array of strings added to the [original] string.
+
+@example {[
+  concatMany "1st" [|"2nd"; "3rd"; "4th"|]  = "1st2nd3rd4th";;
+]}
+*)
+external concatMany : t -> t array -> t = "concat" [@@bs.send] [@@bs.splice]
+
+(** ES2015:
+    [endsWith substr str] returns [true] if the [str] ends with [substr], [false] otherwise.
+    
+@example {[
+  endsWith "BuckleScript" "Script"  = true;;
+  endsWith "BuckleShoes" "Script"  = false;;
+]} 
+*)
+external endsWith : t -> t -> bool = "" [@@bs.send] 
+
+
+(**
+  [includes s searchValue ] returns [true] if [searchValue] is found anywhere within [s], [false] otherwise.
+  
+@example {[
+  includes "programmer" "gram" = true;;
+  includes "programmer" "er" = true;;
+  includes "programmer" "pro" = true;;
+  includes "programmer" "xyz" = false;;
+]}
+*)
+external includes : t -> t -> bool = "" [@@bs.send] (** ES2015 *)
+
+
+
+(**
+  [repeat n s] returns a string that consists of [n] repetitions of [s]. Raises [RangeError] if [n] is negative.
+  
+@example {[
+  repeat "ha" 3 = "hahaha"
+  repeat "empty" 0 = ""
+]}
+*)
+external repeat : t -> int -> t = "" [@@bs.send] (** ES2015 *)
+
+(** [replace string substr newSubstr ] returns a new string which is
+identical to [string] except with the first matching instance of [substr]
+replaced by [newSubstr].
+
+[substr] is treated as a verbatim string to match, not a regular
+expression.
+
+@example {[
+  replace "old string" "old" "new"  = "new string"
+  replace "the cat and the dog" "the" "this"  = "this cat and the dog"
+]}
+*)
+external replace : t -> t ->  t ->  t = "" [@@bs.send]
+
```

Comment:
This is a good place to use label?

Reply:
I can PR and target your branch if you want. 

I'm planning on also adding `Belt.String.length`, `Belt.String.slice/sliceToEnd`, `Belt.String.get`, `Belt.String.indexOf` if that's ok with you. 

should it also include `Belt.String.replaceRegex`, `Belt.String.matchRegex` ou should they just remain in `Js.String`?

Reply:
what is the type of indexOf, I suggest we avoid char type in the API, otherwise it looks good to me

Reply:
the type would be `t -> t -> int option`